### PR TITLE
Feature/hw 6 tests

### PR DESCRIPTION
**Pull request by rosklyar**

### CODE REVIEW

File: garbage_classifier/garbage_classifier/model.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from transformers import BeitForImageClassification
+from sklearn.metrics import f1_score as measure_f1_score
+from tqdm import tqdm
+import torch
+
+def get_model(dropout_rate, fc_layer_size, n_classes):
+    model = BeitForImageClassification.from_pretrained(
+        'microsoft/beit-base-patch16-224-pt22k-ft22k')
```
Comment (truskovskiyk) [inline garbage_classifier/garbage_classifier/model.py]: remove hardcode

File: garbage_classifier/garbage_classifier/model.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    :param dataloader:
+    :return:
+        res: f1_score
+    """
+    print('Model scoring was started...')
+    model.eval()
+    dataloader.dataset.mode = 'eval'
+    result = []
+    targets = []
+    with torch.no_grad():
```
Comment (truskovskiyk) [inline garbage_classifier/garbage_classifier/model.py]: might be just a decorator for a function

File: garbage_classifier/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+numpy
+psutil
+pytest
+scikit_learn
+torch
+torchvision
+tqdm
+transformers
+wandb
```
Comment (truskovskiyk) [inline garbage_classifier/requirements.txt]: add versions

File: garbage_classifier/tests/test_garbage_data.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import torch
+import pytest
+
+from garbage_classifier import garbage_data as gd
+
+
+@pytest.mark.parametrize("input_size", [(3, 256, 256), (3, 512, 512), (3, 1024, 1024), (3, 640, 480)])
+def test_feature_extractor(input_size):
```
Comment (truskovskiyk) [inline garbage_classifier/tests/test_garbage_data.py]: give more clear name: 
`test_feature_extractor_output_shape`

File: garbage_classifier/tests/test_garbage_data.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    feature_extractor = gd.FeatureExtractor(
+        'microsoft/beit-base-patch16-224-pt22k-ft22k')
+    assert feature_extractor is not None
+
+    features = feature_extractor.forward(torch.rand(*input_size))
+    assert features['pixel_values'].shape == torch.Size([1, 3, 224, 224])
+
+
+def test_garbage_data():
```
Comment (truskovskiyk) [inline garbage_classifier/tests/test_garbage_data.py]: give better name please

File: garbage_classifier/tests/test_trainer.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import pytest
+import torch
+from garbage_classifier import trainer as t, model as m
+from garbage_classifier import garbage_data as gd
+
+
+@pytest.fixture
+def data():
+    return gd.GarbageData(
+        "garbage_classifier/tests/data", 2, 0.2)
```
Comment (truskovskiyk) [inline garbage_classifier/tests/test_trainer.py]: looks like we can re-use this fixture from model tests

**Review by truskovskiyk — COMMENTED** [review]
@rosklyar great work, see feedback

File: garbage_classifier/train.py [hunk 7]
```diff
@@ -100,21 +27,30 @@ def train_model(config=None):
...
             wandb.log({'val_f1': f1_score})
             torch.save(model, join(opt.checkpoint_dir,
                        f'epoch{epoch}_f1={round(f1_score, 5)}.pth'))
         # create card
         create_card(config, f1_score)
+        # save model
+        torch.save(model, join(opt.checkpoint_dir, 'weights', 'latest', 'model.pth'))
+        art = wandb.Artifact(f'uwg-garbage-classifier-{wandb.run.id}', type="model")
+        art.add_file(join(opt.checkpoint_dir, 'weights', 'latest', 'model.pth'), "model.pth")
+        art.add_file("garbage_classifier/README.md", "README.md")
```
Comment (truskovskiyk) [inline garbage_classifier/train.py:51]: could we make this a function?

File: garbage_classifier/garbage_classifier/model.py (hunk 1 above)
Comment (rosklyar) [reply garbage_classifier/garbage_classifier/model.py]: done

File: garbage_classifier/garbage_classifier/model.py (hunk 2 above)
Comment (rosklyar) [reply garbage_classifier/garbage_classifier/model.py]: done

File: garbage_classifier/requirements.txt (hunk 3 above)
Comment (rosklyar) [reply garbage_classifier/requirements.txt]: done

File: garbage_classifier/tests/test_garbage_data.py (hunk 4 above)
Comment (rosklyar) [reply garbage_classifier/tests/test_garbage_data.py]: done

File: garbage_classifier/tests/test_garbage_data.py (hunk 5 above)
Comment (rosklyar) [reply garbage_classifier/tests/test_garbage_data.py]: done

File: garbage_classifier/train.py (hunk 7 above)
Comment (rosklyar) [reply garbage_classifier/train.py:51]: done in further prs :)

https://github.com/rosklyar/projector-ml-ops/pull/10/files#diff-d63aebfeb03543fee5807276a22ea2ed95c38661d0613ce71858b1fa4337e86c

**Review by truskovskiyk — APPROVED** [review]

**Pull request closed without merging**